papertty.py

```diff
@@ -475,32 +485,31 @@ def sigusr1_handler(sig, frame):
                 oldbuff = ''
                 flags['scrub_requested'] = False
             with open(vcsa, 'rb') as f:
-                # read the first 4 bytes to get the console attributes
-                attributes = f.read(4)
-                rows, cols, x, y = list(map(ord, struct.unpack('cccc', attributes)))
-
-                # read rest of the console content into buffer
-                buff = f.read()
-                # SKIP all the attribute bytes
-                # (change this (and write more code!) if you want to use the attributes with a
-                # three-color display)
-                buff = buff[0::2]
-                # find character under cursor (in case using a non-fixed width font)
-                char_under_cursor = buff[y * rows + x]
-                cursor = (x, y, char_under_cursor)
-                # add newlines per column count
-                buff = ''.join([r.decode(ptty.encoding, 'ignore') + '\n' for r in ptty.split(buff, cols)])
-                # do something only if content has changed or cursor was moved
-                if buff != oldbuff or cursor != oldcursor:
-                    # show new content
-                    oldimage = ptty.showtext(buff, fill=ptty.black, cursor=cursor if not nocursor else None,
-                                             oldimage=oldimage,
-                                             **textargs)
-                    oldbuff = buff
-                    oldcursor = cursor
-                else:
-                    # delay before next update check
-                    time.sleep(float(sleep))
+                character_width, vcsudev = ptty.vcsudev(vcsa)
+                with open(vcsudev, 'rb') as vcsu:
+                    # read the first 4 bytes to get the console attributes
+                    attributes = f.read(4)
+                    rows, cols, x, y = list(map(ord, struct.unpack('cccc', attributes)))
+
+                    # read from the text buffer 
+                    buff = vcsu.read()
```

Comment:
Not sure yet if we actually want to do this but adding it here for reference: this seems to fix the `tmux` issue I had.

```suggestion
                    buff = vcsu.read()
                    # fix tmux whitespace
                    if character_width == 4:
                        buff = buff.replace(b'\x20\x20\x20\x20', b'\x20\x00\x00\x00')

```